packages/gatsby-source-sensenet/README.md

```diff
@@ -0,0 +1,411 @@
+        Name
+        DisplayName
+      }
+    }
+  }
+}
+```
+
+### Create pages
+
+When building pages from contents you may want to create URL for your blogpost contents. E.g. if you have a content with markdown body at /Root/Content/SampleWorkspace/Blog/2021-02-23-how-we-post, you might want to turn that into a page on your site at example.com/2021-02-23-how-we-post/.
+
+This is an example how can you implement:
+
+When "sensenetBlogPost" node is created you should create a new field with the name of "slug" on the node:
+
+```javascript
+//In your gatsby-node.js
+exports.onCreateNode = async ({ node, actions: { createNodeField } }) => {
+  if (node.internal.type === 'sensenetBlogPost') {
+    createNodeField({
+      node,
+      name: 'slug',
+      value: node.Name || '',
+    })
+  }
+}
+```
+
+Then iterate through all sensenetBlogPost and create page with using a template for all.
+
+```javascript
+//In your gatsby-node.js
+exports.createPages = async ({ graphql, actions }) => {
+  const { createPage } = actions
+
+  const allSensenetBlogPost = await graphql(`
+    {
+      allSensenetBlogPost(limit: 1000) {
+        edges {
+          node {
+            fields {
+              slug
+            }
+          }
+        }
+      }
+    }
+  `)
+
+  if (allSensenetBlogPost.errors) {
+    console.error(allSensenetBlogPost.errors)
+    throw new Error(allSensenetBlogPost.errors)
+  }
+
+  const blogPostTemplate = path.resolve('<YOUR TEMPLATE>')
+
+  allSensenetBlogPost.data.allSensenetBlogPost.edges.forEach(({ node }) => {
+    const { slug } = node.fields
+
+    createPage({
+      path: `/${slug}/`,
+      component: blogPostTemplate,
+      context: {
+        slug,
+      },
+    })
+  })
+}
+```
+
+## Complex types
+
+### Image
+
+In sensenet images can be references. In this case the LeadImage field of a Blogpost will be something like this:
+
+```json
+"LeadImage": {
+  "__deferred": {
+    "uri": "/odata.svc/Root/Content/SampleWorkspace/Blog('2021-01-13-docviewer-updates')/LeadImage"
+  }
+},
+```
+
+Reference fields have to be expanded. You can define the fields you would like expand in the gatsby-config by listing the fieldNames in an array, eg.:
+`oDataOptions: {expand: ['LeadImage']},`
+
+Then you can create a remote file node from it:
+
+```javascript
+exports.onCreateNode = async ({ node, actions: { createNode }, createNodeId, getCache }) => {
+  if (node.internal.type === 'sensenetBlogPost') {
+    const leadImageNode = await createRemoteFileNode({
+      url: `<YOUR REPOSITORY URL>${node.LeadImage.Path}`,
+      parentNodeId: node.Id.toString(),
+      createNode,
+      createNodeId,
+      getCache,
+    })
+    if (leadImageNode) {
+      node.leadImage___NODE = leadImageNode.id ///connect to blogpost node
+    }
+  }
+}
+```
+
+<note severity="info">The content should be public otherwise you have to provide a httpHeaders: { Authorization: `Bearer <yourAccessToken>` } attribute to createRemoteFileNode function</note>
```

Comment:
```suggestion
<note severity="info">If you want to request content from a sensenet repository, they should be public otherwise you have to provide a httpHeaders: { Authorization: `Bearer <yourAccessToken>` } attribute to createRemoteFileNode function</note>
```